### PR TITLE
More engine update funcs

### PR DESCRIPTION
**Pull request by joshlory**
Trying to discover the rest of the shape of `EngineObj` and whether `D_80141BDC` is an array or struct

### CODE REVIEW

File: src/main/323C.c [hunk 1]
```diff
@@ -2116,17 +2129,167 @@ INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_800204CC);
...
+
+    if (D_80141BDC[0] == 0) {
+        switch (D_80141BDF[0]) {
+        case 0:
+            if (D_80171EA8 != 0) {
+                func_800127C8(1);
+                func_80013014();
+                D_80171EA8 = 0;
+            }
+            if (engine_obj.unk5A & PADLleft) {
```
Comment (sozud) [inline src/main/323C.c:2158]: Doesn't seem clear that this is a copy of the pad data or not but this is not an issue for now

File: src/main/323C.c [hunk 2]
```diff
@@ -2116,17 +2129,167 @@ INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_800204CC);
...
-INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_80020A08);
+    if (D_80141BDC[0] == 0) {
+        if (D_80141BDF[0] == 0) {
+            if (D_80171EA8 != 0) {
+                func_800127C8(1);
+                func_80013014();
+                D_80171EA8 = 0;
+            }
+
+            if (engine_obj.unk5A & 0x8000) {
```
Comment (sozud) [inline src/main/323C.c:2276]: If you are sure it's pad data there's still a couple of spots without macros

**Review by sozud — APPROVED** [review]
Great work!

File: src/main/323C.c (hunk 1 above)
Comment (joshlory) [reply src/main/323C.c:2158]: Whoops you're right

**Pull request merged**